### ISSUE TITLE
Getting x509: certificate has expired or is not yet valid in 1.3.0 istio

### ISSUE THREAD

**Issue by piyushkumar13** [istio/istio#20478]
I started getting following error all of a sudden while creating virtualservice and destinationrule(which was working fine earlier): 
```
Internal error occurred: failed calling webhook "pilot.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitpilot?timeout=30s: x509: certificate has expired or is not yet valid
```
I am using following istio configuration : 
client version: 1.3.0
control plane version: 1.3.0

Installed istio using default values.yaml and following command : 
```
helm template install/kubernetes/helm/istio \
--set gateways.istio-ingressgateway.type=NodePort \
--set grafana.enabled=true \
--set tracing.enabled=true \
--set kiali.enabled=true \
--set global.controlPlaneSecurityEnabled=true \
--set global.mtls.enabled=true \
--name istio --namespace istio-system | kubectl apply -f -
```
I checked the root certificate expiration date by following doc : https://istio.io/docs/ops/configuration/security/root-transition/#root-transition-procedure It shows certificate is valid for approximately 10 years. Here is the output : 
```
Fetching root cert from istio-system namespace...
Your Root Cert will expire after
   Sep 24 06:26:51 2029 GMT
Current time is
  Fri Jan 24 17:12:38 IST 2020


=====YOU HAVE 3530 DAYS BEFORE THE ROOT CERT EXPIRES!=====
```

I saw this issue as closed https://github.com/istio/istio/issues/14517. According to this, it should get resolved in `1.3.0`

**Comment by piyushkumar13** [comment]
I checked the root certificate expiration date by following doc : https://istio.io/docs/ops/configuration/security/root-transition/#root-transition-procedure It shows certificate is valid for approximately 10 years. Here is the output : 
```
Fetching root cert from istio-system namespace...
Your Root Cert will expire after
   Sep 24 06:26:51 2029 GMT
Current time is
  Fri Jan 24 17:12:38 IST 2020


=====YOU HAVE 3530 DAYS BEFORE THE ROOT CERT EXPIRES!=====
```
But, I am getting this error while creating istio virtualservice and destinationrules which were working fine earlier
```
Internal error occurred: failed calling webhook "pilot.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitpilot?timeout=30s: x509: certificate has expired or is not yet valid
```

**Comment by taylorreece** [comment]
Maybe it's coincidence, but the `samples/certs/*` certs were created on 2018-01-24.  It's precisely two years later.  The certs do *not* expire today, though, but maybe some sample 2-year certs were created on the same day or something?

**Comment by taylorreece** [comment]
@piyushkumar13 OOC if you
```
get secret istio.istio-galley-service-account -n istio-system -o yaml
```
And then snag the `cert-chain.pem` and run 
```
echo 'TheBigLongEncryptedKeyYouFound==' | base64 -d
```
And throw that into
https://www.sslshopper.com/certificate-decoder.html,
Is your istio-galley certificate still valid?

**Comment by piyushkumar13** [comment]
@taylorreece I tried with what you mentioned. The certificate is still valid. Here is the info on `cert-chain.pem` : 
```
Certificate Information:
Subject Alternative Names: URI:spiffe://cluster.local/ns/istio-system/sa/istio-galley-service-account, istio-galley.istio-system.svc, istio-galley.istio-system
Valid From: December 25, 2019
Valid To: March 24, 2020
Issuer: cluster.local
Serial Number: 36956f64e98f66510049e47514b35bf0
```

> with a clean install of istio 1.3.0, I'm not seeing the same issues that you are seeing. I can launch a virtualservice / deployment just fine.
It was working fine for me as well. I was able to create virtualservice and destinationrules as well. But, I started getting the error all of a sudden.

> Looks like #17718 is more recent and was fixed by #17995, which is not in 1.3.0, but would likely be in 1.3.6 or 1.4.x

I can try updating to 1.3.6V or 1.4v. But, wanted to root cause what has caused this issue in `1.3.0`

**Comment by piyushkumar13** [comment]
@taylorreece After upgrading to `1.4` issue resolved.

**Comment by alexvolsh** [comment]
I have the same issue on a fresh install of the Istio `1.4.3` with enabled SDS. When SDS disabled, everything works fine.

**Comment by piyushkumar13** [comment]
I am closing it since issue resolved after updating to `1.4` version.

**Comment by GiannisSialmas** [comment]
Hi guys, i dont know if my problem is relevant with this issue, but this is the issue that clicked most.

I currently have istio 1.1.0 installed on my production cluster. We have implemented an authorization adapter as a seperate microservice, to which we apply a rules.config.istio.io resource in which we describe which requests should pass from the authorization.

When i went to edit the rule to whitelist a url route, after saving i got the following error in my console.
```
error: rules.config.istio.io "r1" could not be patched: Internal error occurred: failed calling webhook "mixer.validation.istio.io": Post https://istio-galley.istio-system.svc:443/admitmixer?timeout=30s: x509: certificate has expired or is not yet valid
```
My galley logs at that time showed the following
```
2020-05-12T09:55:10.125506Z	info	http: TLS handshake error from 10.132.0.125:55982: remote error: tls: bad certificate
2020-05-12T09:55:10.138130Z	info	http: TLS handshake error from 10.132.0.101:39314: remote error: tls: bad certificate
```
By doing some googling, i run the following command 
```
kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io istio-galley -ojsonpath="{.webhooks[0].clientConfig.caBundle}" | base64 -d > /tmp/cert-inj && openssl x509 -noout -text -in /tmp/cert-inj
```
from which i got that a certificate is not valid after May 6(currently 6 days in the past)
```
Certificate:
    Data:
        Version: 3 (0x2)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: O = cluster.local
        Validity
            Not Before: May  7 12:53:30 2019 GMT
            Not After : May  6 12:53:30 2020 GMT
```
Is the only way of solving this an istio upgrade to version 1.4? We are planning on doing it at some point but not in the very immediate future as it will also require a lot of other items to be done first.
The most important from them being changing our authorization as from what i understand, mixer (and as a result adapters) are deprecated in future version of istio.

Thanks in advance
